head_ref/src/HeadReference.cpp

```diff
@@ -275,6 +338,61 @@ bool HeadReference::targetToPanTilt(const tf::Stamped<tf::Point>& target, double
     return true;
 }
 
+
+bool HeadReference::getJointsInfo()
+{
+  urdf::Model model;
+  if (!model.initParam(tf_prefix_ + "/robot_description"))
+  {
+    ROS_ERROR("Cannot get URDF model from the parameter server");
+    return false;
+  }
+
+  if (!getJointInfo(model, pan_joint_props_))
+    return false;
+
+  if (!getJointInfo(model, tilt_joint_props_))
+    return false;
+
+  return true;
+
+}
+
+
+bool HeadReference::getJointInfo(const urdf::Model &model, JointProps &props)
+{
+  // Get joint info from urdf model
+  urdf::JointConstSharedPtr joint_ptr = model.getJoint(props.name);
+  if (joint_ptr == NULL)
+  {
+    ROS_ERROR_STREAM("Could not get joint '" << props.name << "' from the urdf model");
+    return false;
+  }
+
+  // Get joint limits from joint info
+  props.lower = joint_ptr->limits->lower;
+  props.upper = joint_ptr->limits->upper;
+
+  // Get direction from joint info
+  if (joint_ptr->axis.x > 0.99 || joint_ptr->axis.y > 0.99 || joint_ptr->axis.z > 0.99)
+    props.direction = 1.0;
+  else if (joint_ptr->axis.x < -0.99 || joint_ptr->axis.y < -0.99 || joint_ptr->axis.z < -0.99)
+    props.direction = -1.0;
+  else
+  {
+    ROS_ERROR_STREAM("Don't know ho to define the direction of joint " << props.name);
```

Comment:
*how

Reply:
Error message could be clearer, I would have no idea what to fix in case of this error. 
It's debatable how often it will occur though...

Reply:
Update in 103fe7631ef3a41eae9f55ca56547b37282e97d7